### PR TITLE
4 create flight endpoint

### LINKED ISSUE #4: Create Flight Endpoint
`POST /api/v1/users/1/flights`
Required Params:
`Night Hours, Day Hours, Aircraft, Description, Date, Start Location, End Location, Role`
```

{
    "data": {
        "id": 1,
        "type": "flight",
        "attributes": {
            "night_hours": "2",
            "day_hours": "1",
            "aircraft": "Boeing 737",
            "description": "great time!",
            "date": "2023-03-09",
            "start_location": "DEN",
            "end_location": "LAX",
            "role": "pilot"
        }
    }
}
```

### PR DESCRIPTION
**Pull request by brenicillin**
## Neccesary checkmarks:

    [x] All Tests are Passing in all environments

    [x] The code will run locally

## Type of change

    [x] New feature
    [] Bug Fix
    [] Refactor

## Description of Change:

    description: 
    
## Requested feedback: 
    
    I'd like feedback on... 

## Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

## Testing Changes

    [x] No Tests have been changed
    [] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:

    [x] My code has no unused/commented out code
    [x] My code has no breakpoints
    [x] I have reviewed my code
    [] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch

Link:
https://media.tenor.com/isrmdo9Xr9EAAAAC/mission-accomplished.gif

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,4 @@
-.venv
\ No newline at end of file
+.venv
+*.pyc
```
Comment (KatieLonsdale) [inline .gitignore:2]: if you want to untrack all these pycache files so Git ignores them, [check this out.](https://gist.github.com/mcrd25/1b2718054b65f4720fa582c6a4032217)

File: flight_log_be/models.py [hunk 2]
```diff
@@ -8,15 +9,15 @@ def __str__(self):
...
+  day_hours = models.DecimalField(max_digits=4, decimal_places=2, default=0.00)
+  night_hours = models.DecimalField(max_digits=4, decimal_places=2, default=0.00)
+  aircraft = models.CharField(max_length=50, blank=False)
+  description = models.CharField(max_length=500, blank=True)
+  role = models.CharField(max_length=50, blank=False)
 
   def __str__(self):
-    return str(self.start_location + " to " + self.end_location)
\ No newline at end of file
+    return '{} to {}'.format(self.start_location, self.end_location)
```
Comment (KatieLonsdale) [inline flight_log_be/models.py:23]: Nice.

File: flight_log_be/serializers.py [hunk 3]
```diff
@@ -1,7 +1,20 @@
 from rest_framework import serializers
 from .models import User
+from .models import Flight
 
 class UserSerializer(serializers.ModelSerializer):
     class Meta:
         model = User
-        fields = ['id', 'first_name', 'last_name']
\ No newline at end of file
+        fields = ['id', 'first_name', 'last_name']
```
Comment (KatieLonsdale) [inline flight_log_be/serializers.py]: I removed id on my branch, just a head's up for merge conflicts.

File: flight_log_be/tests/test_views.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            'end_location': 'Los Angeles',
+            'day_hours': 5,
+            'night_hours': 3,
+            'description': 'Test flight',
+        }
+    
+        response = self.client.post(f'/api/v1/users/{user.id}/flights/',dict(flight_data))
+        
+        self.assertEqual(response.status_code, 201)
+
```
Comment (KatieLonsdale) [inline flight_log_be/tests/test_views.py:23]: I would just add some tests that the response includes the keys/values we want.

**Review by KatieLonsdale — APPROVED** [review]
Good job!

File: .gitignore (hunk 1 above)
Comment (cariperi) [reply .gitignore:2]: Cool!

**Review by cariperi — APPROVED** [review]
Nicely done! Looking forward to handling some merge conflicts and maybe a conditional for the get / post routes when we merge mine!

**Pull request merged**